### DISCUSSION TITLE
Default Value

### DISCUSSION THREAD

**Discussion by bumpyy** [joshhanley/livewire-autocomplete#6] [Help] [answered]
Is there a way to preselect value based on id ?.

e.g : when i use it on edit form, so the data already prefilled.

**Comment by dircm** [comment, marked as answer]
You sure can!  Say you were working with client models.  To use the autocomplete component, you need to provide the text, ID and results as separate public properties in your parent livewire component.  So for example, prior to rendering, you would set `$this->clientID = clientModel->id` and  `$this->clientName = clientModel->title`.  You just need to configure the component with the pre-loaded properties like below.  

```php
<x-lwa::autocomplete
    name="my-autocomplete"
    wire:model-id="clientId"
    wire:model-text="clientName"
    wire:model-results="clients"
    wire:focus="getClients"
```

**Comment by joshhanley** [comment]
@bumpyy yeah @dircm is correct, you can preselect a value by loading the value from your selected model into whatever property you have bound to `wire:model-text`, like `clientName` above.

**Comment by bumpyy** [comment]
Thank's, that make sense, so i just have to query the user by id in the mount lifecycle hook right ?

**Reply by joshhanley** [reply]
@bumpyy yep to populate it initially. You can also change it's value whenever you need by doing that in an action method or updated hook too 🙂
